.github/workflows/LINUX_BUILD_TEST.yml

```diff
@@ -0,0 +1,736 @@
+      #######################################
+      - name: INFO
+        shell: bash
+        run: |
+          cat /etc/os-release 
+          echo "_______________________________________"
+          uname -a
+          echo "_______________________________________"
+          cmake --version
+          echo "_______________________________________"
+          "${{ env.C_COMPILER }}" --version
+          which gcc
+          echo "_______________________________________"
+          "${{ env.CPP_COMPILER }}" --version
+          which g++
+          echo "_______________________________________"
+          
+      #######################################
+      # Run Integration Tests.
+      #######################################
+      - name: Integration Tests
+        shell: bash
+        run: |
+          ./build/test/integration/integrationtests
+
+      #######################################
+      # Run Unit Tests.
+      #######################################
+      - name: Unit Tests
+        shell: bash
+        run: |
+          ./build/test/unit/unittests
+
+      #######################################
+      # Run Regression AARCH64 Tests.
+      #######################################
+      - name: Regression Test (aarch64)
+        if: always()
+        shell: bash
+        run: |
+          ./build/test/regression/aarch64/regression-aarch64
+
+      #######################################
+      # Run Regression RISCV Tests.
+      #######################################
+      - name: Regression Test (riscv)
+        if: always()
+        shell: bash
+        run: |
+          ./build/test/regression/riscv/regression-riscv
+
+      #######################################
+      # Run Benchmarks
+      #######################################
+
+      - if: ${{ contains(fromJson('["ubuntu:18.04"]'), matrix.OS) && inputs.SIMENG-MODE == 'Release' }}
+        name: Checking out benchmark repository (v3)
+        uses: actions/checkout@v3
+        with:
+          repository: UoB-HPC/simeng-benchmarks
+          token: ${{ env.PAT }}
+          ref: makefile-build-system
+          path: simeng-benchmarks
+
+      - if: ${{ !contains(fromJson('["ubuntu:18.04"]'), matrix.OS) && inputs.SIMENG-MODE == 'Release' }}
+        name: Checking out benchmark repository (v4)
+        uses: actions/checkout@v4
+        with:
+          repository: UoB-HPC/simeng-benchmarks
+          ref: makefile-build-system
+          token: ${{ env.PAT }}
+          path: simeng-benchmarks
+
+      - if: ${{ inputs.SIMENG-MODE == 'Release' }}
+        name: CloverLeaf serial gcc8.3.0 armv8.4
+        uses: ./.github/actions/simeng_benchmarks
+        with:
+          RUN_DIR: $GITHUB_WORKSPACE/simeng-benchmarks/Data_Files/CloverLeaf
+          BIN_PATH: $GITHUB_WORKSPACE/simeng-benchmarks/binaries/CloverLeaf/serial/cloverleaf_gcc8.3.0_armv8.4
+          PASS_STRING: "This test is considered PASSED"
+
+      - if: ${{ inputs.SIMENG-MODE == 'Release' }}
```

Comment:
Can these instead be one command per benchmark, generic for compiler? If you name the binary correctly (with compiler in), then you could just use the variable of the compiler in the bin path? That way, adding a new benchmark takes one block for all OS/compilers which is much nicer imo as this is **all** repeated. Same in `MACOS_BUILD_TEST`.

Reply:
Seeing one of the resolved comments, I understand now why it's all separated. Can you please add a comment here and in `MACOS_BUILD_TEST` as to why they're separated.